src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,344 @@
+                            , carouselLabel = config.name
+                            , announceAndSelect = config.announceAndSelect
+                            }
+                    }
+    in
+    { viewPreviousButton = viewPreviousButton
+    , viewNextButton = viewNextButton
+    , slides =
+        Html.div [ Accessibility.Styled.Key.tabbable False ]
+            (List.map
+                (\slide ->
+                    Html.div
+                        [ Role.group
+                        , Aria.roleDescription "slide"
+                        , id slide.idString
+                        , labelAttribute slide
+
+                        -- use as attribute for testing
+                        , if config.selected == slide.id then
+                            Attrs.style "display" "block"
+
+                          else
+                            Attrs.style "display" "none"
+                        ]
+                        [ slide.slideHtml ]
+                )
+                config.slides
+            )
+    , containerAttributes =
+        [ Attrs.attribute "role" (roleToString config.role)
+        , Aria.roleDescription "carousel"
+        , labelAttribute config
+        ]
+    }
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+
+  - `controls`: tabs control buttons
+  - `slides` container with the carousel contents
+
+-}
+viewWithTabControls :
+    { selected : id
+    , slides :
+        List
+            { id : id
+            , idString : String
+            , name : String
+            , visibleLabelId : Maybe String
+            , slideHtml : Html msg
+            , tabControlHtml : Html Never
+            }
+    , tabControlStyles : Bool -> List Style
+    , tabControlListStyles : List Style
+    , role : Role
+    , name : String
+    , visibleLabelId : Maybe String
+    , focusAndSelect : { select : id, focus : Maybe String } -> msg
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithTabControls config =
+    let
+        buildTab tabConfig =
+            TabsInternal.fromList { id = tabConfig.id, idString = tabConfig.idString }
+                [ \tab -> { tab | panelView = tabConfig.slideHtml }
+                , \tab -> { tab | tabView = [ Html.map never tabConfig.tabControlHtml ] }
+                ]
+
+        { tabList, tabPanels } =
+            TabsInternal.views
+                { focusAndSelect = config.focusAndSelect
+                , selected = config.selected
+                , tabs = List.map buildTab config.slides
+                , tabStyles = always config.tabControlStyles
+                , tabListStyles = config.tabControlListStyles
+                }
+    in
+    { controls = tabList
+    , slides = tabPanels
+    , containerAttributes =
+        [ Attrs.attribute "role" (roleToString config.role)
+        , Aria.roleDescription "carousel"
+        , labelAttribute config
+        ]
```

Comment:
So the previous version of Carousel didn't return `containerAttributes` from `Carousel.view` -- it just returned `controls` and `slides`.

Do we need this return value? If so, could you add it to the documentation please?

If not, then I think there are some values in the API for this function that aren't necessary anymore:
- role
- name
- visibleLabelId

Reply:
Oh, I see -- this is the same topic as https://github.com/NoRedInk/noredink-ui/pull/1488/files#r1306015638.

Let's add the return value to the documentation then, if you would! 🙏